framework/render/tasks.py

```diff
@@ -39,36 +40,42 @@ def render_mfr_error(err):
         """.format(**locals())
 
 
+# TODO only allow one task at a time
 @app.task(ignore_result=True, timeout=settings.MFR_TIMEOUT)
-def _build_rendered_html(file_path, cache_dir, cache_file_name, download_url):
+def _build_rendered_html(download_url, cache_path, temp_path):
     """
     :param str file_path: Full path to raw file on disk
     :param str cache_dir: Folder to store cached file in
     :param str cache_file_name: Name of cached file
     :param str download_url: External download URL
     """
-    with codecs.open(file_path) as file_pointer:
+    try:
+        ensure_path(os.path.split(temp_path)[0])
+        urllib.urlretrieve(download_url, temp_path)
+    except:
```

Comment:
Only catch specific exception(s). Also, does `ensure_path` need to be in the `try` block?